Use try-with-resources where possible.

cmnbroad opened the pull request:
Note that I didn't do anything with places where either htsjdk.samtools.util.CloseUtil or org.apache.commons.io.deleteQuietly are used, although both seem like future candidates for inspection.

akiezun commented:
@droazen please review

cmnbroad commented:
@droazen Can you take a look at these changes. Thx.

akiezun commented:
i'll take it

akiezun commented on src/main/java/org/broadinstitute/hellbender/tools/picard/sam/FilterReads.java:
```diff
@@ -112,18 +112,17 @@ private void filterReads(final FilteringIterator filteringIterator) {
-
-        bw.close();
-        reader.close();
         IOUtil.assertFileIsReadable(readsFile);
```
this line needs a null check for readsFile (in case an exception happens before readsFile is initialized to a file object)

cmnbroad replied:
Good point - it should probably just be moved inside the try block.

akiezun commented on src/test/java/org/broadinstitute/hellbender/tools/exome/ExomeReadCountIntegrationTest.java:
```diff
@@ -524,9 +524,10 @@ private void compareTableFiles(File outputFile, File expectedOutputFile, final S
-            fr.close();
+            Table result = null;
+            try (final FileReader fr = new FileReader(file)) {
+                result = new Table(fr);
```
Can you just say `return new Table(fr)` inside of the try block? Or will the compiler complain?

akiezun commented on src/test/java/org/broadinstitute/hellbender/tools/picard/vcf/AbstractVcfMergingClpTester.java:
```diff
@@ -153,13 +155,13 @@ private void validateResultsForMultipleInputs(final File output, final List<Queu
-		reader.close();
-		return contigPositions;
+        final Queue<String> contigPositions = new LinkedList<>();
+		try (final VCFFileReader reader = new VCFFileReader(inputFile, false)) {
```
can there 2 blocks be combined into 1?

akiezun replied:
if not, put in a comment why not? (because whoever reads this code will wonder)

akiezun commented:
looks good. small fixes to make. back to @cmnbroad. when done, please reasign to me to re-review

akiezun commented:
(nevermind, i can't assign to @cmnbroad because of silly github). Just ping me when done

cmnbroad commented:
back to @akiezun

akiezun commented:
are these two problems fixed? I can't tell because the new commits are incrementally. Is there a way to sync your master to ours and rebase the branch on top of that? If I merge now it'll be a bit of a mess.

cmnbroad commented:
I fixed the problems, squashed the commits, pulled upstream master and rebased on that. But it doesn't  look right - there is an unrelated commit in there. I'll probably just apply these changes to a new branch and do a PR on that just to be safe. I'll let you know when its ready to look at.
